remus/common/ExecuteProcess.h

```diff
@@ -58,12 +58,39 @@ class REMUSCOMMON_EXPORT ExecuteProcess
   //and wasn't set to run in detached mode
   virtual ~ExecuteProcess();
 
+  //Additional command and its args. The process's input pipe will be the output
+  //pipe of the preceding process.
+  void appendProcess(const std::string& command, const std::vector<std::string>& args);
+
+  //Additional command. The process's input pipe will be the output pipe of the
+  //preceding process.
+  void appendProcess(const std::string& command);
+
   //execute the process.
   virtual void execute();
 
   //kills the process if running
   virtual bool kill();
 
+  //Set whether the given pipe in the child is shared with the parent process.
+  //The default is no for Pipe_STDOUT and Pipe_STDERR and yes for Pipe_STDIN.
+  void sharePipeWithParent(ProcessPipe::PipeType pipe, bool choice);
```

Comment:
I would prefer for ```sharePipeWithParent``` to be private to force the usage of the more explicit syntax.